app/scripts/modules/canary/acaTask/acaTaskStage.js

```diff
@@ -59,16 +59,28 @@ module.exports = angular.module('spinnaker.canary.acaTaskStage', [
         : $scope.stage.canary.watchers //if it is not an array it is probably a SpEL
       : '';
 
-    accountService.getUniqueAttributeForAllAccounts('aws', 'regions')
-      .then( (regions) => {
-        $scope.regions = regions.sort();
-      });
+    var applicationProviders = $scope.application.attributes.cloudProviders;
+    $scope.accounts = new Array();
+    $scope.regions = new Array();
 
+    if (applicationProviders.length === 0) {
+      applicationProviders[0] = 'aws'; // default to AWS if no provider is set
+    }
 
-    accountService.listAccounts('aws').then(function(accounts) {
-      $scope.accounts = accounts;
-    });
+    for (var i = 0; i < applicationProviders.length; i++) {
+      if (applicationProviders[i] === 'aws') {
+        accountService.getUniqueAttributeForAllAccounts(applicationProviders[i], 'regions')
+        .then((regions) => {
+          $scope.regions = regions.sort();
```

Comment:
If I'm reading this correctly, regions will only be available from AWS accounts. Is that intentional? I can see that making sense for someone running AWS and Kubernetes, but what about GCP, which will have different regions?